medusa/server/api/v1/core.py

```diff
@@ -1797,8 +1797,8 @@ def run(self):
                 raise ApiError('Status Prohibited')
             app.STATUS_DEFAULT = self.status
 
-        if self.flatten_folders is not None:
-            app.FLATTEN_FOLDERS_DEFAULT = int(self.flatten_folders)
+        if self.season_folders is not None:
+            app.FLATTEN_FOLDERS_DEFAULT = int(not self.season_folders)
```

Comment:
As we discussed on Slack, the `app` variable name should change too:
`FLATTEN_FOLDERS_DEFAULT => SEASON_FOLDERS_DEFAULT`